flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -185,6 +187,18 @@
 	private volatile boolean running = true;
 
 	// ------------------------------------------------------------------------
+	//  internal metrics
+	// ------------------------------------------------------------------------
+
+	/** Counter for successful Kafka offset commits. */
+	private transient Counter successfulCommits;
+
+	/** Counter for failed Kafka offset commits. */
+	private transient Counter failedCommits;
+
+	private transient KafkaCommitCallback offsetCommitCallback;
```

Comment:
Sounds fair, I'll include a javadoc as well a notice about the thread-safety contract as you suggested.